sdk/textanalytics/Azure.AI.TextAnalytics/src/DocumentHealthcareEntities.cs

```diff
@@ -0,0 +1,117 @@
+        /// <param name="statistics"> if showStats=true was specified in the request this field will contain information about the document payload. </param>
+        internal DocumentHealthcareEntities(string id, IReadOnlyList<HealthcareEntity> entities, IReadOnlyList<HealthcareRelationInternal> relations, IReadOnlyList<TextAnalyticsWarningInternal> warnings, TextDocumentStatistics? statistics)
+        {
+            Id = id;
+            Entities = entities;
+            Relations = ResolveHealthcareRelations(entities, relations);
+            Warnings = Transforms.ConvertToWarnings(warnings);
+            Statistics = statistics;
+        }
+
+        internal static IReadOnlyList<HealthcareRelation> ResolveHealthcareRelations(IEnumerable<HealthcareEntity> entities, IEnumerable<HealthcareRelationInternal> relations)
+        {
+            List<HealthcareRelation> list = new List<HealthcareRelation>();
+            foreach (HealthcareRelationInternal relation in relations)
+            {
+                list.Add(new HealthcareRelation(relation.RelationType,
+                    relation.Bidirectional,
+                    ResolveHealthcareEntity(entities, relation.Source),
+                    ResolveHealthcareEntity(entities, relation.Target)));
+            }
+
+            return list;
+        }
+
+        /// <summary> Unique, non-empty document identifier. </summary>
+        public string Id { get; }
+        /// <summary> Healthcare entities. </summary>
+        public IReadOnlyList<HealthcareEntity> Entities { get; }
+        /// <summary> Healthcare entity relations. </summary>
+        public IReadOnlyList<HealthcareRelation> Relations { get; }
+        /// <summary> Warnings encountered while processing document. </summary>
+        public IReadOnlyList<TextAnalyticsWarning> Warnings { get; }
+        /// <summary> if showStats=true was specified in the request this field will contain information about the document payload. </summary>
+        public TextDocumentStatistics? Statistics { get; }
+
+        private static Regex _healthcareEntityRegex = new Regex(@"\#/results/documents\/(?<documentIndex>\d*)\/entities\/(?<entityIndex>\d*)$", RegexOptions.Compiled, TimeSpan.FromSeconds(2));
+
+        internal static HealthcareEntity ResolveHealthcareEntity(IEnumerable<HealthcareEntity> entities, string reference)
+        {
+            var healthcareEntityMatch = _healthcareEntityRegex.Match(reference);
+            if (healthcareEntityMatch.Success)
+            {
+                int entityIndex = int.Parse(healthcareEntityMatch.Groups["entityIndex"].Value, CultureInfo.InvariantCulture);
+                //int entityIndex = int.Parse(healthcareEntityMatch.Groups[2].Value, CultureInfo.InvariantCulture);
+
+                if (entityIndex < entities.Count())
+                {
+                    var entity = entities.ElementAt(entityIndex);
+                    return new HealthcareEntity(entity.Text, entity.Category,
```

Comment:
Looks like `entities` is already a `list<HealthcareEntity>`, why do you create a new `HealthcareEntity`?

Reply:
updated